### PR TITLE
[Nano] API for searching best environment variables dynamically with `subprocess`

### PR DESCRIPTION
**Pull request by hoshibara**
## Description

Some models performed poorly with default environment variables, even 50% slower than with other environment variables.

`spawn_new_process` was introduced in #7025 , but `LD_PRELOAD` has no effect on new process.

That's why I implemented `bigdl.nano.utils.inference.common.utils.exec_with_worker`. It uses `subprocess.run` to create new process for testing different environments, and uses files to pass arguments.

### CODE REVIEW

File: python/nano/src/bigdl/nano/utils/common/optimizer/optimizer.py [hunk 1]
```diff
@@ -160,5 +172,49 @@ def get_best_model(self,
...
+            self._search_env(method_name=best_metric.method_name)
+
         return best_model, format_acceleration_option(best_metric.method_name,
                                                       self.ALL_INFERENCE_ACCELERATION_METHOD)
+
+    def _search_env(self, method_name: str = None):
+        invalidInputError(method_name,
+                          'Must give specific model method for searching env.')
+        invalidInputError(method_name in self.optimized_model_dict,
+                          'Must provide correct method name.')
```
Comment (TheaperDeng) [inline python/nano/src/bigdl/nano/utils/common/optimizer/optimizer.py:185]: method_name should be one of {}, but get {}.

File: python/nano/src/bigdl/nano/utils/common/optimizer/optimizer.py [hunk 2]
```diff
@@ -160,5 +172,49 @@ def get_best_model(self,
...
+            for env_name, latency in env_result_map.items():
+                print(f"{env_name}\t{latency} ms")
+            best_env = self.ALL_ACCELERATION_ENV[
+                min(env_result_map, key=env_result_map.get)].get_env_dict()
+            self.optimized_model_dict[method_name]['env'] = best_env
+        env_suggestion = '\n'.join(f'export {env_key}=\'{env_value}\''
+                                   for env_key, env_value in
+                                   self.optimized_model_dict[method_name]['env'].items())
+        register_suggestion(f'You can try the following commands for better performance\n'
+                            f'{env_suggestion}')
```
Comment (TheaperDeng) [inline python/nano/src/bigdl/nano/utils/common/optimizer/optimizer.py:210]: the suggestion could be dump to a sh file as we talked about, but we may do it in next PR.

File: python/nano/src/bigdl/nano/pytorch/inference/optimizer.py [hunk 3]
```diff
@@ -13,8 +13,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+import tempfile
 
 import torch
+from bigdl.nano.utils.common.optimizer.acceleration_env import AccelerationEnv
+from bigdl.nano.utils.common.optimizer.exec_with_worker import exec_with_worker
```
[the remark this answers is not in the document]
Comment (MeouSker77) [reply python/nano/src/bigdl/nano/pytorch/inference/optimizer.py:20]: yes,

File: python/nano/src/bigdl/nano/utils/common/optimizer/acceleration_env.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        else:
+            tmp_env_dict['LD_PRELOAD'] = ''
+            tmp_env_dict['MALLOC_CONF'] = ''
+
+        # set omp env var
+        omp_lib_path = ''
+        if AccelerationEnv._CONDA_DIR:
+            if os.path.exists(os.path.join(AccelerationEnv._CONDA_DIR, '../lib/libiomp5.so')):
+                omp_lib_path = os.path.join(AccelerationEnv._CONDA_DIR, '../lib/libiomp5.so')
+            elif os.path.exists(os.path.join(AccelerationEnv._CONDA_DIR,
```
[the remark this answers is not in the document]
Comment (yangw1234) [reply python/nano/src/bigdl/nano/utils/common/optimizer/acceleration_env.py:78]: @TheaperDeng 
Do you think it is possible to move all library searching and environmental variable setting into one python module? And maybe `source bigdl-nano-init` can also call that module.

**Pull request closed without merging**